crux_core/src/typegen.rs

```diff
@@ -298,3 +343,35 @@ fn copy(from: impl AsRef<Path>, to: impl AsRef<Path>) -> Result<()> {
 
     Ok(())
 }
+
+#[cfg(feature = "typegen")]
+#[cfg(test)]
+mod tests {
+    use crate::typegen::{Samples, TypeGen};
+    use serde::{Deserialize, Serialize};
+    use uuid::Uuid;
+
+    #[derive(Serialize, Deserialize)]
+    struct MyUuid(Uuid);
+
+    #[test]
+    fn test_typegen_for_uuid_without_samples() {
```

Comment:
I don't know if this test is ultimately useful, but I found it helpful for validating the change, and thought it might be useful in review. We can remove it before merging.